src/Search/FlagrowIOSearcher.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Flagrow\Bazaar\Search;
+
+use Flagrow\Bazaar\Extensions\Extension;
+use Flarum\Core\Search\SearchResults;
+use Flarum\Extension\ExtensionManager;
+use Flarum\Settings\SettingsRepositoryInterface;
+use GuzzleHttp\Client;
+use Illuminate\Database\Eloquent\Collection;
+
+class FlagrowIOSearcher extends AbstractExtensionSearcher
+{
+    /**
+     * @var ExtensionManager
+     */
+    protected $extensionManager;
+
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $config;
+
+    /**
+     * @param ExtensionManager $manager
+     * @param SettingsRepositoryInterface $config
+     */
+    public function __construct(ExtensionManager $manager, SettingsRepositoryInterface $config)
+    {
+        $this->extensionManager = $manager;
+        $this->config = $config;
+    }
+
+    /**
+     * Get a Guzzle client configured for Flagrow API
+     * @return Client
+     */
+    protected function getClient()
+    {
+        return new Client([
+            'base_uri' => 'http://localhost:8000/api/',
+            'headers' => [
+                'Accept' => 'application/json',
+                'Authorization' => 'Bearer '.$this->config->get('flagrow.bazaar.api_token'),
+            ]
+        ]);
+    }
+
+    /**
+     * Create an Extension object and map all data sources
+     * @param array $apiPackage
+     * @return Extension
+     */
+    public function createExtension(array $apiPackage)
+    {
+        $extension = Extension::createFromAttributes($apiPackage['attributes']);
+
+        $installedExtension = $this->extensionManager->getExtension($extension->getShortName());
+
+        if (!is_null($installedExtension)) {
+            $extension->setInstalledExtension($installedExtension);
+        }
+
+        return $extension;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function search($limit = null, $offset = 0)
+    {
+        $responseHtml = $this->getClient()->get('packages', [
+            'json' => [
+                'page' => $offset,
+                //'offset' => $offset,
+                //'limit' => $limit,
+            ],
+        ]);
+
+        $responseJson = json_decode($responseHtml->getBody(), true);
+
+        $packages = $responseJson['data'];
+        $pagination = $responseJson['meta']['pagination'];
+
+        $areMoreResults = $pagination['total_pages'] > $pagination['current_page'];
+
+        $extensions = new Collection();
```

Comment:
I had to use Eloquent Collection to be compatible with Flarum `SearchResults` class. This was apparently not meant to be used with anything else than database...

Maybe there are more appropriate Contracts to use ?